gcloud-java-dns/src/main/java/com/google/gcloud/dns/spi/DefaultDnsRpc.java

```diff
@@ -54,6 +62,26 @@ public DefaultDnsRpc(DnsOptions options) {
   }
 
   @Override
+  public BatchRequest enqueueListZones(BatchRequest batch, final DnsBatch.Request request,
```

Comment:
Your reasoning is sound. My point is that I would like to see all the `DnsBatch` -> `BatchRequest` conversion happen in the same place but as you outlined this is probably not attainable. Fine as is for me, thanks for taking the time to explain your reasoning.